src/TreeNode.tsx

```
@@ -231,12 +231,12 @@ class InternalTreeNode extends React.Component<InternalTreeNodeProps, TreeNodeSt
   };
 
   isDisabled = () => {
-    const { disabled } = this.props;
+    const { disabled, data } = this.props;
     const {
-      context: { disabled: treeDisabled },
+      context: { disabled: treeDisabled, disabledStrategy },
     } = this.props;
 
-    return !!(treeDisabled || disabled);
+    return !!(treeDisabled || disabled || (disabledStrategy && disabledStrategy(data)));
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)